sessionctx/variable/sysvar.go

```diff
@@ -1399,6 +1399,10 @@ var defaultSysVars = []*SysVar{
 		s.AssertionLevel = tidbOptAssertionLevel(val)
 		return nil
 	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBBatchPendingTiFlashCount, Value: strconv.Itoa(DefTiDBBatchPendingTiFlashCount), MinValue: 1, MaxValue: math.MaxUint32, Hidden: false, Type: TypeUnsigned, SetSession: func(s *SessionVars, val string) error {
+		s.AssertionLevel = tidbOptAssertionLevel(val)
```

Comment:
This is obviously wrong. You could add a new field in sctx, and move the logic of `getBatchPendingTiFlashCount` here. So that you could `sctx.vars.BatchPendingTiFlashCount`

But that means it will not take effect instantly like a global variable. All session variables are not instantly-effected variables.